### PR TITLE
SeqManager.cpp: fix num discontinued bug and increase performance

### PR DESCRIPTION
**Pull request by penguinol**
#395

### CODE REVIEW

**Review by jmillan — APPROVED** [review]

**Review by ibc — COMMENTED** [review]
The test does not compile for me. It looks like you are not synced with `v3` branch:

```
../test/src/RTC/TestSeqManager.cpp:409:15: error: use of class template 'TestSeqManagerInput' requires template
      arguments
                std::vector<TestSeqManagerInput> inputs =
```

**Comment by ibc** [comment]
Please take a look to the current `TestSeqManager.cpp` in mediasoup `v3` branch:

* https://github.com/versatica/mediasoup/blob/v3/worker/test/src/RTC/TestSeqManager.cpp

All `inputs` are created with the proper template syntax:

```c++
std::vector<TestSeqManagerInput<uint16_t>> inputs =
```

However the diff in your PR does not include those templates.

**Comment by ibc** [comment]
Also, I think `SeqMaanger.cpp` should include `#include <iterator>` which exposes `std::distance()`.

**Comment by ibc** [comment]
> Also, I think `SeqMaanger.cpp` should include `#include <iterator>` which exposes `std::distance()`.

Please add this and we are done. And let me know your name/URL or something to add into mediasoup Ackknowledges.

File: worker/src/RTC/SeqManager.cpp [hunk 1]
```diff
@@ -1,6 +1,7 @@
 #define MS_CLASS "RTC::SeqManager"
 // #define MS_LOG_DEV_LEVEL 3
 
+#include <iterator>
```
Comment (ibc) [inline worker/src/RTC/SeqManager.cpp:4]: This must be at the end. Otherwise `make lint` will fail. `make format` will fix it for you.

Comment (ibc) [reply worker/src/RTC/SeqManager.cpp:4]: Will do myself, don't worry.

**Comment by ibc** [comment]
Merged, thanks a lot!

**Comment by penguinol** [comment]
> > Also, I think `SeqMaanger.cpp` should include `#include <iterator>` which exposes `std::distance()`.
> 
> Please add this and we are done. And let me know your name/URL or something to add into mediasoup Ackknowledges.

JunLong Zou
www.lonny.me

**Comment by ibc** [comment]
Thanks, added here: https://mediasoup.org/about/#acknowledgements

**Pull request merged**